tools/memap.py

```diff
@@ -205,18 +253,41 @@ def parse_map_file_gcc(self, file_desc):
                 elif change_section != False:
                     current_section = change_section
 
-                [module_name, module_size] = self.parse_section_gcc(line)
+                [object_name, object_size] = self.parse_section_gcc(line)
 
-                if module_size == 0 or module_name == "":
+                if object_size == 0 or object_name == "":
                     pass
                 else:
-                    self.module_add(module_name, module_size, current_section)
+                    self.module_add(object_name, object_size,\
+                                        current_section)
+
+    def parse_object_name_armcc(self, line):
+        """ Parse object file
+
+        Positional arguments:
+        line - the line containing the object or library
+        """
+
+        # simple object (not library)
+        if line[-3] == '.' and line[-2] == 'o':
+            object_name = line[:-1]
+            return object_name
+
+        else:
+
+            rex_obj_name = r'(.+\.l)\((.+\.o)\)'
```

Comment:
This comment is the same as the other Regular expression related ones:
1) `re` not `rex` is common in python
2) Please make this a Class or global constant